lindi/LindiH5ZarrStore/LindiH5ZarrStore.py

```diff
@@ -402,39 +449,59 @@ def to_file(self, file_name: str, *, file_type: Literal["zarr.json"] = "zarr.jso
     def to_reference_file_system(self) -> dict:
         """Create a reference file system cooresponding to this store.
 
-        This can then be loaded using
-        LindiClient.from_reference_file_system(obj)
+        This can then be loaded using LindiH5pyFile.from_reference_file_system(obj)
         """
         if self._h5f is None:
             raise Exception("Store is closed")
         if self._url is None:
             raise Exception("You must specify a url to create a reference file system")
         ret = {"refs": {}, "version": 1}
 
+        # TODO: use templates to decrease the size of the JSON
+
         def _add_ref(key: str, content: Union[bytes, None]):
+            import base64
             if content is None:
                 raise Exception(f"Unable to get content for key {key}")
             try:
-                ret["refs"][key] = content.decode("ascii")
+                if content.startswith(b"base64:"):
+                    # This is the rare case where the content actually starts with "base64:"
+                    # which is confusing. Not sure when this would happen, but it could.
+                    ret["refs"][key] = (b"base64:" + base64.b64encode(content)).decode(
+                        "ascii"
+                    )
+                else:
+                    # This is the usual case. It will raise a UnicodeDecodeError if the
+                    # content is not valid ASCII, in which case the content will be
+                    # base64 encoded.
+                    ret["refs"][key] = content.decode("ascii")
             except UnicodeDecodeError:
-                import base64
-
-                # from kerchunk
+                # If the content is not valid ASCII, then we base64 encode it. The
+                # reference file system reader will know what to do with it.
                 ret["refs"][key] = (b"base64:" + base64.b64encode(content)).decode(
                     "ascii"
                 )
```

Comment:
Super minor - but should the try/except be around line 477? If the decode error happens on 470, it will just happen again in the except block because the lines are the same.

Reply:
good point, I made the change